### PR TITLE
Always whitelist the login fallback for SSO

### PR DESCRIPTION
**Pull request by babolivier**
That fallback sets the redirect URL to itself (so it can process the login token then return gracefully to the client). This would make it pointless to ask the user for confirmation, since the URL the confirmation page would be showing wouldn't be the client's.

### CODE REVIEW

File: docs/sample_config.yaml [hunk 1]
```diff
@@ -1392,7 +1392,11 @@ sso:
     # phishing attacks from evil.site. To avoid this, include a slash after the
     # hostname: "https://my.client/".
     #
-    # By default, this list is empty.
+    # If public_baseurl is set, then the login fallback page (used by clients
+    # that don't have full support for SSO) is always included in this list.
+    #
+    # By default, this list is empty, except if public_baseurl is set (in which
+    # case the login fallback page is the only element in the list).
```
Comment (clokep) [inline docs/sample_config.yaml]: I believe that `public_basurl` *has* to be set for SAML to work. Not sure if this changes the comment or not (it does not have to be set for CAS).

Comment (babolivier) [reply docs/sample_config.yaml]: It doesn't, this config is for both CAS and SAML so we don't expect `base_url` to be always whitelisted.

Comment (richvdh) [inline docs/sample_config.yaml]: this seems like a complicated way to explain this.

How about: 

> If public_baseurl is set, then the login fallback page (used by clients
> that don't natively support the required login flows) is whitelisted in addition
> to any URLs listed here.
>
> By default, this list is empty.

Comment (richvdh) [reply docs/sample_config.yaml]: or possibly reverse those two paragraphs, which I think makes it clearer which "list" we're talking about.

Comment (babolivier) [reply docs/sample_config.yaml]: Done (with a slight re-phrasing) in https://github.com/matrix-org/synapse/pull/7153/commits/63aea691a761a9b6a2058b54792fc5859e12cfba

**Comment by richvdh** [comment]
errr this should't have gone into master. will revert and merge to develop instead.

**Comment by babolivier** [comment]
Ugh, crap, sorry. Can't wait for develop to be the main branch. Thanks for handling.

**Pull request merged**